src/meta/assessment/description/nationalDataDataSourceDescription.ts

```diff
@@ -0,0 +1,17 @@
+import { Label } from '@meta/assessment'
+
+export interface DataSourceVariable {
+  value: string
+  label: Label
+  prefixLabel?: Label
+}
+
+export interface DataSourceDescriptionTable {
+  variables: Array<DataSourceVariable>
+  typeOfDataSourceText?: boolean
```

Comment:
here it should be intuitive to understand that the type of input can depend on a predefined array of variables, or text input if no variables are provided. 

To make it easier, i would suggest to make variables an optional property.
The component will behave this way: if array of variables is provided, then values change based on the select; 
otherwise, a text input.

What do you think?

Reply:
This makes sense

What about typeOfDataSourceText? I was struggling to find a better name, so left it open for discussion

This boolean, when set to true, renders a text input field, otherwise a dropdown select of the following:

( https://github.com/openforis/fra-platform/blob/7a2d1f9744cd4060d13fe6a2d24159af6dbc9ad4/src/meta/assessment/description/dataSourceType.ts )

Reply:
I don't think we need that typeOfDataSourceText  property at all.

we could render dropdown if array of variables is provided, otherwise render text input.

what do you think?

Reply:
This is for types, not variables (second column, refer to video)

Reply:
sure, sounds good

ill handle the first comment